### ISSUE TITLE
Using image loaded from another thread

### ISSUE THREAD

**Issue by baehyunsol** [not-fl3/macroquad#604]
Is it okay to use an image (`Texture2D`) loaded from another thread? I spawned 2 processes, A and B. I called `Texture2D::from_file_with_format` in process A, and sent the `Texture2D` from the function to process B using [`mpsc::Sender`](https://doc.rust-lang.org/stable/std/sync/mpsc/struct.Sender.html).

My code is working but it's showing an empty image. I suspect something's wrong with the `mpsc` stuff. Is it safe to send `Texture2D` across processes?

**Comment by not-fl3** [comment]
No, macroquad silently assumes that all its API calls are made from one single thread (like we have on wasm).
 
Unfortunately macroquad is not yet panicing on incorrent usage :(

**Comment by baehyunsol** [comment]
In that case, what would be the best way to use a `Texture2D` created from another thread? Is there any workaround?

I want them to run on different processes because I don't want the resource loading to block the main game process.

**Comment by baehyunsol** [comment]
Okay, I should figure out another way to load resources background

Thanks!
